src/test/java/se/nbis/lega/inbox/sftp/InboxSftpEventListenerTest.java

```diff
@@ -122,6 +125,31 @@ public void renameFile() throws IOException {
         assertEquals(DigestUtils.sha256Hex(FileUtils.openInputStream(file)), encryptedIntegrity.getChecksum());
     }
 
+    @Test
+    public void renameFolder() throws IOException {
+        sftpClient.mkdir("test");
+        sftpClient.mkdir("test/test1");
+
+        sftpClient.put(file.getAbsolutePath(), "test/test1/" + file.getName());
+        fileBlockingQueue.poll();
+
+        sftpClient.rename("test/test1", "test/test2");
+
+        FileDescriptor fileDescriptor = fileBlockingQueue.poll();
+        assertNotNull(fileDescriptor);
+        assertEquals(username, fileDescriptor.getUser());
+        String expectedOldPath = inboxFolder + "/" + username + "/test/test1";
+        String expectedPath = inboxFolder + "/" + username + "/test/test2";
+        assertTrue(new File(expectedPath).exists());
```

Comment:
Use Junit rule `ErrorCollector` 
https://carminetech.wordpress.com/2015/10/30/junit-how-to-collect-multiple-assertion-results-in-one-test/